tests/contract_testcases/solana/error2.dot

```diff
@@ -0,0 +1,50 @@
+strict digraph "tests/contract_testcases/solana/error2.sol" {
+	X1 [label="name: X1\ntests/contract_testcases/solana/error2.sol:2:1-14\nvalue: a"]
+	X2 [label="name: X2\ntests/contract_testcases/solana/error2.sol:7:1-14\nvalue: b"]
+	X2_4 [label="name:X2\ntests/contract_testcases/solana/error2.sol:6:7-9\nfield name:bar ty:int256"]
+	X2_5 [label="name:X2\ncontract: 0\ntests/contract_testcases/solana/error2.sol:11:8-10\nfield name:bar ty:int256"]
+	X3 [label="name:X3\ncontract: 0\ntests/contract_testcases/solana/error2.sol:12:8-10\nfield name:X ty:int256"]
+	X4 [label="name:X4\ncontract: 0\ntests/contract_testcases/solana/error2.sol:13:8-10\nfield name:X ty:int256\nfield name: ty:bool"]
+	X5 [label="name:X5\ncontract: 0\ntests/contract_testcases/solana/error2.sol:14:8-10\nfield name: ty:unresolved"]
+	X6 [label="name:X6\ncontract: 0\ntests/contract_testcases/solana/error2.sol:15:8-10\nfield name: ty:unresolved"]
+	contract [label="contract C\ntests/contract_testcases/solana/error2.sol:10:1-18:2"]
+	diagnostic [label="X1 is already defined as an enum\nlevel Error\ntests/contract_testcases/solana/error2.sol:3:7-9"]
+	note [label="location of previous definition\ntests/contract_testcases/solana/error2.sol:2:6-8"]
+	diagnostic_15 [label="expecting 'error', got 'foo'\nlevel Error\ntests/contract_testcases/solana/error2.sol:4:1-4"]
+	diagnostic_16 [label="expression found where 'error' was expected\nlevel Error\ntests/contract_testcases/solana/error2.sol:5:1-4"]
+	diagnostic_17 [label="X2 is already defined as an error\nlevel Error\ntests/contract_testcases/solana/error2.sol:7:6-8"]
+	note_18 [label="location of previous definition\ntests/contract_testcases/solana/error2.sol:6:7-9"]
+	diagnostic_19 [label="found contract 'C'\nlevel Debug\ntests/contract_testcases/solana/error2.sol:10:1-18:2"]
+	diagnostic_20 [label="X2 is already defined as an error\nlevel Warning\ntests/contract_testcases/solana/error2.sol:11:8-10"]
+	note_21 [label="location of previous definition\ntests/contract_testcases/solana/error2.sol:6:7-9"]
+	diagnostic_22 [label="error 'X3' has duplicate field name 'X'\nlevel Error\ntests/contract_testcases/solana/error2.sol:12:23-24"]
+	note_23 [label="location of previous declaration of 'X'\ntests/contract_testcases/solana/error2.sol:12:11-16"]
+	diagnostic_24 [label="mapping type is not permitted as error field\nlevel Error\ntests/contract_testcases/solana/error2.sol:14:11-32"]
+	diagnostic_25 [label="type 'X' not found\nlevel Error\ntests/contract_testcases/solana/error2.sol:15:11-12"]
+	diagnostic_26 [label="expecting 'error', got 'meh'\nlevel Error\ntests/contract_testcases/solana/error2.sol:16:2-5"]
```

Comment:
Were we expecting an error? A function definition in line 16 could be valid.

Reply:
I don't know what you mean.

`error` is not a keyword. The parser generates a parse tree like `<expression> <name>(<fields>..)`. The first `expression` should be the identifier `error`.

Reply:
I am not sure what the developers intent was when we get in this situation. They may have misspelt `error`, or they may have been trying to do somthing completely different. I am not sure.

Reply:
```solidity
	error X6(X);
	meh X7();
```
In this example, `meh` is not unexpected construct, but is does not mean we are expecting another error declaration. I could have been an event. I think that saying `expected 'error'` in the message is misleading in this sense.

Reply:
What should the diagnostic be then?

Reply:
How about `declaration expected, but found 'meh' instead`?

Reply:
There is another case:
```solidity
contract c {
       int foo(int bar);
}
```
I am not sure if there is much we can say other than `expecting 'function', 'error', or 'event'`

Reply:
> This looks like a good option.

I don't think it makes much sense, because it assumes that the developer's intention was to declare a variable and that seems unlikely - why would it have  parameters then.

Reply:
What's your suggestion?

Reply:
> What's your suggestion?

I can only think of generic phrases right now, like `incorrect declaration`.